cmd/format-config-v1.go

```diff
@@ -469,29 +453,47 @@ func findDiskIndex(disk string, jbod []string) int {
 	return -1
 }
 
-// reorderDisks - reorder disks in JBOD order.
-func reorderDisks(bootstrapDisks []StorageAPI, formatConfigs []*formatConfigV1) ([]StorageAPI, error) {
-	var savedJBOD []string
-	for _, format := range formatConfigs {
-		if format == nil {
-			continue
+// reorderDisks - reorder disks in JBOD order, and return reference
+// format-config
+func reorderDisks(bootstrapDisks []StorageAPI,
+	formatConfigs []*formatConfigV1) (*formatConfigV1, []StorageAPI,
+	error) {
+
+	// Pick first non-nil format-cfg as reference
+	var refCfg *formatConfigV1
+	for _, formatConfig := range formatConfigs {
+		if formatConfig != nil {
+			refCfg = formatConfig
+			break
 		}
-		savedJBOD = format.XL.JBOD
-		break
 	}
-	// Pick the first JBOD list to verify the order and construct new set of disk slice.
+	if refCfg == nil {
+		return nil, nil, fmt.Errorf("could not find any valid config")
+	}
+	refJBOD := refCfg.XL.JBOD
+
+	// construct reordered disk slice
 	var newDisks = make([]StorageAPI, len(bootstrapDisks))
 	for fIndex, format := range formatConfigs {
 		if format == nil {
 			continue
 		}
-		jIndex := findDiskIndex(format.XL.Disk, savedJBOD)
+		jIndex := findDiskIndex(format.XL.Disk, refJBOD)
 		if jIndex == -1 {
-			return nil, errors.New("Unrecognized uuid " + format.XL.Disk + " found")
+			return nil, nil, errors.New("Unrecognized uuid " + format.XL.Disk + " found")
 		}
 		newDisks[jIndex] = bootstrapDisks[fIndex]
 	}
-	return newDisks, nil
+
+	// Based on orderedDisks generate new UUIDs in the ref. config
+	// for disks without format-configs.
+	for index, disk := range newDisks {
```

Comment:
This was done to improve code sharing between `healFormatXL*()` funcs, but this function is also called from `loadFormatXL()` which has no need of the assigning UUIDs. So now I've added an extra flag to skip this step in case of loadFormatXL. The function is not renamed, as it would be too long to call it something like `reorderDisksAndAssignUUIDsToUnassignedDisks` - but the extra flag is not documented in the function description.